draft-ietf-mls-protocol.md

```diff
@@ -4872,27 +4885,72 @@ handle extensible fields:
+* A client joining a group MUST verify that it supports every extension in the
+  GroupContext for the group.  Otherwise, it should treat the enclosing
+  GroupInfo message as invalid and not join the group.
 
-Note that the latter two requirements mean that all MLS extensions are
-mandatory, in the sense that an extension in use by the group MUST be supported
-by all members of the group.
+Note that the latter two requirements mean that all MLS GroupContext extensions
+are mandatory, in the sense that an extension in use by the group MUST be
+supported by all members of the group.
 
 The parameters of a group may be changed by sending a GroupContextExtensions
-proposal to enable additional extensions, or by reinitializing the group as
-described in {{reinitialization}}.
+proposal to enable additional extensions ({{groupcontextextensions}}), or
+by reinitializing the group ({{reinitialization}}).
+
+## GREASE
+
+As described in {{extensions}}, clients are required to ignore unknown values
+for certain parameters.  To help ensure that other clients implement this
+behavior, a client can follow the “Generate Random Extensions And Sustain
+Extensibility” or GREASE approach described in {{?RFC8701}}.  In the context of
+MLS, this means that a client creating a LeafNode or Welcome message includes
+random values in certain fields, which should be ignored by a
+correctly-implemented client processing the message.  A client that incorrectly
+rejects unknown code points will fail to process such a message, providing a
+signal to its implementor that the client needs to be fixed.
+
+When generating the following fields, an MLS client SHOULD include a random
+selection of values chosen from these GREASE values:
+
+* `LeafNode.capabilities.versions`
+* `LeafNode.capabilities.ciphersuites`
+* `LeafNode.capabilities.extensions`
+* `LeafNode.capabilities.proposals`
+* `LeafNode.capabilities.credentials`
+* `KeyPackage.extensions`
+* `GroupInfo.extensions`
+
+For the KeyPackage and GroupInfo extensions, the `extension_data` for GREASE
+extensions MAY have any contents selected by the sender, since they will be
+ignored by a correctly-operating receiver.  For example, a senders might
+populate these extensions with a variable-sized amount of random data.
+
+A set of values reserved for GREASE have been registered in the various
+registries in {{iana-considerations}}.  This prevents conflict between GREASE
+and real future values.  The following values are reserved in each registry:
+`0x0A0A`, `0x1A1A`, `0x2A2A`, `0x3A3A`, `0x4A4A`, `0x5A5A`, `0x6A6A`, `0x7A7A`,
+`0x8A8A`, `0x9A9A`, `0xAAAA`, `0xBABA`, `0xCACA`, `0xDADA`, and `0xEAEA`.  (The
+value `0xFAFA` falls within the private use range.). These values MUST only
+appear in the fields listed above, and not, for example, in the `proposal_type`
+field of a Proposal.  Clients MUST NOT implement special processing for these
+values, since this negates their utility for detecting extensibility failures.
```

Comment:
```suggestion
field of a Proposal.  Clients MUST NOT implement any special processing rules 
for how to handle these values when receiving them, since this negates their
utility for detecting extensibility failures.
```